kong/api/routes/certificates.lua

```diff
@@ -1,317 +1,80 @@
-      end
-
-      snis_in_request[sni] = true
-    end
-  end
-
-  local ssl_cert, err = dao_factory.ssl_certificates:insert(self.params)
-  if err then
-    return helpers.yield_error(err)
-  end
-
-  ssl_cert.snis = setmetatable({}, cjson.empty_array_mt)
-
-  -- insert SNIs if given
-
-  if snis then
-    for i, sni in ipairs(snis) do
-      local ssl_server_name = {
-        name                = sni,
-        ssl_certificate_id  = ssl_cert.id,
-      }
-
-      local row, err = dao_factory.ssl_servers_names:insert(ssl_server_name)
-      if err then
-        return helpers.yield_error(err)
-      end
-
-      ssl_cert.snis[i] = row.name
-    end
-  end
-
-  return helpers.responses.send_HTTP_CREATED(ssl_cert)
-end
-
-
-local function update_certificate(self, dao_factory, helpers)
-   -- check if exists
-   local ssl_cert, err = dao_factory.ssl_certificates:find {
-    id = self.params.id
-  }
-  if err then
-    return helpers.yield_error(err)
-  end
-
-  if not ssl_cert then
-    return helpers.responses.send_HTTP_NOT_FOUND()
-  end
-
-  local snis
-  if type(self.params.snis) == "string" then
-    snis = utils.split(self.params.snis, ",")
-
-  elseif self.params.snis == ngx.null then
-    snis = {}
-  end
-
-  self.params.snis = nil
-
-  local snis_in_request = {} -- check for duplicate snis in the request
-  local snis_in_db = {} -- avoid db insert if sni is already present in db
-
-  -- if snis field present
-  -- 1. no duplicate snis should be present in the request
-  -- 2. check if any sni in the request is using a cert
-  --    other than the one being updated
-
-  if snis then
-    for _, sni in ipairs(snis) do
-      if snis_in_request[sni] then
-        return helpers.responses.send_HTTP_CONFLICT("duplicate SNI in " ..
-                                                    "request: " .. sni)
-      end
-
-      local sni_in_db, err = dao_factory.ssl_servers_names:find {
-        name = sni,
-      }
-      if err then
-        return helpers.yield_error(err)
-      end
-
-      if sni_in_db then
-        if sni_in_db.ssl_certificate_id ~= ssl_cert.id then
-          return helpers.responses.send_HTTP_CONFLICT(
-            "SNI '" .. sni .. "' already associated with existing " ..
-            "certificate (" .. sni_in_db.ssl_certificate_id .. ")"
-          )
-        end
-
-        snis_in_db[sni] = true
-      end
-
-      snis_in_request[sni] = true
-    end
-  end
-
-  local old_snis, err = dao_factory.ssl_servers_names:find_all {
-    ssl_certificate_id = ssl_cert.id,
-  }
-  if err then
-    return helpers.yield_error(err)
-  end
-
-  -- update certificate if necessary
-  if self.params.key or self.params.cert then
-    self.params.created_at = ssl_cert.created_at
-
-    ssl_cert, err = dao_factory.ssl_certificates:update(self.params, {
-      id = self.params.id,
-    }, { full = true })
-    if err then
-      return helpers.yield_error(err)
+local function get_cert_by_sni_or_id(self, db, helpers)
```

Comment:
What this function does is changing the params called `certificates` (as in `/certificates/:certificates`). If the param looks like a uuid it does nothing. Otherwise, it is assumed that the param is an sni name. If the sni for that name is found, the `certificates` param takes the value of the sni's `certificate_id`.

I have renamed the function to `get_cert_id_from_sni`, probably this will make it a bit more clear.